packages/vuetify/src/labs/rules/rules.ts

```diff
@@ -0,0 +1,108 @@
+// Composables
+import { useLocale } from '@/composables'
+
+// Utilities
+import { inject } from 'vue'
+import { mergeDeep } from '@/util'
+
+// Types
+import type { InjectionKey } from 'vue'
+import type { ValidationRule } from '@/composables/validation'
+
+export type ValidationRuleBuilderWithOptions<T> = (options: T, err?: string) => ValidationRule
+export type ValidationRuleBuilder = (err?: string) => ValidationRule
+
+export interface RulesInstance {
+  required: ValidationRuleBuilder
+  email: ValidationRuleBuilder
+  number: ValidationRuleBuilder
+  integer: ValidationRuleBuilder
+  capital: ValidationRuleBuilder
+  maxLength: ValidationRuleBuilderWithOptions<number>
+  minLength: ValidationRuleBuilderWithOptions<number>
+  strictLength: ValidationRuleBuilderWithOptions<number>
+  exclude: ValidationRuleBuilderWithOptions<string[]>
+  notEmpty: ValidationRuleBuilder
+  pattern: ValidationRuleBuilderWithOptions<RegExp>
+}
+
+export type InternalRulesOptions = {
+  // No options for now
+}
+
+export type RulesOptions = Partial<InternalRulesOptions>
+
+const parseRulesOptions = (options?: RulesOptions) => {
+  return mergeDeep({
+  }, options)
+}
+
+export function createRules (options?: RulesOptions) {
+  // @ts-expect-error
+  const _options = parseRulesOptions(options)
+
+  const state: RulesInstance = {
+    required: (err?: string) => {
+      const { t } = useLocale()
```

Comment:
You should only need to instantiate this once

Reply:
Are you talking about `t` ?

Reply:
Yes

Reply:
The `locale` dependency being provided after the call of `createRules()` in `framework.ts`, it's not possible to get and share the same `t` instance using `useLocale()` by calling it at the beginning of the `createRules()`;

> [Vuetify] Could not find injected locale instance

Once this is said: Is it really a problem to get `t` in each rule function ? `t` is provided using the Vue's provide/inject (aka dependency injection pattern) system so, the same instance is re-used every time.

Do you really care about it ?

Reply:
Like I said, you *should* only need to. If it's causing an issue otherwise, maybe it's something we need to look into in addition to this change.

Reply:
any review about last changes ?

Reply:
Looking good to you ?